apphub/one_shot_learning/siamese_network_tf.py

```diff
@@ -0,0 +1,192 @@
+import tempfile
+
+import cv2
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.optimizers import Adam
+from tensorflow.python.keras import Model, Sequential, layers
+from tensorflow.python.keras.initializers import RandomNormal
+from tensorflow.python.keras.regularizers import l2
+
+import fastestimator as fe
+from fastestimator.backend import feed_forward
+from fastestimator.dataset import omniglot
+from fastestimator.op.numpyop import Minmax, ReadImage, Sometimes, ShiftScaleRotate
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.trace.adapt import LRScheduler, EarlyStopping
+from fastestimator.trace.io import BestModelSaver
+from fastestimator.trace.metric import Accuracy
+from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
+
+
+def lr_schedule(epoch):
+    lr = 0.0001 * np.power(0.99, epoch)
+    return lr
+
+
+class OneShotAccuracy(Trace):
+    def __init__(self, dataset, model, N=20, trials=400, mode=["eval", "test"], output_name="one_shot_accuracy"):
+
+        super().__init__(mode=mode, outputs=output_name)
+        self.dataset = dataset
+        self.model = model
+        self.total = 0
+        self.correct = 0
+        self.output_name = output_name
+        self.N = N
+        self.trials = trials
+
+    def on_epoch_begin(self, data: Data):
+        self.total = 0
+        self.correct = 0
+
+    def on_epoch_end(self, data: Data):
+        for _ in range(self.trials):
+            img_path = self.dataset.one_shot_trial(self.N)
+            input_img = (np.array([np.expand_dims(cv2.imread(i, cv2.IMREAD_GRAYSCALE), -1) / 255. for i in img_path[0]],
```

Comment:
instead of redefining all the the pre-processing stuff and then calling feed_forward, why not use self.system.network.transform()?

Reply:
oh wait, I guess what you really need would be pipeline. should we add that to system?